### PR TITLE
Indicative pricing calculation tweak

### PR DESCRIPTION
**Pull request by DrMaxis**
### Description

Update the indicative pricing helper to properly calculate the indicative price based on the bounds.
 
- Replaced the condition to use the price value as a high or low bound if the margin is undefined or zero, with a condition to check if the weighted values of the bid are defined and if so, calculate the adjusted price. 

### Notes

- For bids created before the addition of the stoplight, the margin of error condition stands.

### CODE REVIEW

File: src/domain/services/IndicativePricingHelper.js [hunk 1]
```diff
@@ -24,30 +24,27 @@ export default class IndicativePricingHelper {
...
+    let adjustedPrice;
+    if (weights) {
+      adjustedPrice = isLow
+        ? value * (weights[6] / this.bid.cost)
+        : value * (weights[1] / this.bid.cost);
+    } else {
+      if (marginOfError <= 0) return value;
     }
+    return adjustedPrice;
+  }
```
Comment (rreynier) [inline src/domain/services/IndicativePricingHelper.js:47]: Can we write a test for this?

Comment (DrMaxis) [reply src/domain/services/IndicativePricingHelper.js:47]: Yes, we can! I'll start writing one.

Comment (DrMaxis) [reply src/domain/services/IndicativePricingHelper.js:45]: The latest changes have the unit test for the IndicativePricingHelper.

File: src/domain/services/IndicativePricingHelper.js [hunk 2]
```diff
@@ -58,6 +53,7 @@ export default class IndicativePricingHelper {
             this._indicativePricingField = _.find(this.bid.entities.fields(), function(el) {
                 return el.title.toLowerCase() === "indicative pricing";
             });
+            console.log('fields', this.bid.entities.fields());
```
Comment (rreynier) [inline src/domain/services/IndicativePricingHelper.js:56]: Remove please

File: src/domain/services/IndicativePricingHelper.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import Helpers from "../../utils/Helpers";
+const _ = require('lodash');
```
Comment (rreynier) [inline src/domain/services/IndicativePricingHelper.js:3]: Is this needed? Does not appear we are using this helper library anymore.

File: tests/indicative_pricing_helper.spec.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  let cacheValues = [10,20,30,40,50,60,70,80];
+  expect(helper.getIndicativePrice(10, 5, false, 0.5, cacheValues)).toBe(40);
+});
+
+
+
+
+
+
+
```
Comment (rreynier) [inline tests/indicative_pricing_helper.spec.js:32]: Lots of empty lines

**Review by rreynier — APPROVED** [review]
Few minor comments, please resolve and let's merge!

File: src/domain/services/IndicativePricingHelper.js (hunk 2 above)
Comment (DrMaxis) [reply src/domain/services/IndicativePricingHelper.js]: Removed.

File: src/domain/services/IndicativePricingHelper.js (hunk 3 above)
Comment (DrMaxis) [reply src/domain/services/IndicativePricingHelper.js]: Removed.

File: tests/indicative_pricing_helper.spec.js (hunk 4 above)
Comment (DrMaxis) [reply tests/indicative_pricing_helper.spec.js]: Removed.

**Pull request merged**